packages/jaeger-ui/src/components/TracePage/CriticalPath/utils/getChildOfSpans.tsx

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Span } from '../../../../types/trace';
+
+// This function removes child spans whose refType is FOLLOWS_FROM
+const getChildOfSpans = (spans: Span[]): Span[] => {
```

Comment:
same question: https://github.com/jaegertracing/jaeger-ui/pull/1532/files#r1292502031

Reply:
Because in `transformTraceData` have both childOf and followsFrom child spans. So we to get only childOf spans we are using getChildOfSpans function

Reply:
You can use the tree and full list of children already constructed by `transformTraceData` and trim it to needed references. That is still a lot faster than doing linear search across all spans as you're doing.

Reply:
Or even faster - when going through all children in LFC function, skip the FOLLOWS ones. This way you do everything on one pass, without creating tons of additional collections which take memory allocations and degrade performance.

Reply:
Ok I got your idea.
But please clarify my doubt.
  * Can a 'followsFrom' span have 'childOf' spans as children or any descendants?
  
 I am asking this because if we are not considering followsFrom spans in CP algorithm we should also not consider its children or its descendants in CP algorithm.
 So our aim is to not only drop/skip followsFrom spans but also all its descendants.
 getChildOfSpans function not only drops followsFrom spans but also drop all its descendants.
 https://github.com/jaegertracing/jaeger-ui/blob/7a51ede7803c432825b29066e3a31e80cb2f8ef2/packages/jaeger-ui/src/components/TracePage/CriticalPath/utils/getChildOfSpans.tsx#L31-L34 
 
 So should I perform the same logic in transformTraceData to trim the data instead of skipping the spans in LFC function.

Reply:
Yes, FOLLOWS span can have regular CHILD spans.

Shouldn't your algorithm be recursive? E.g. when we ask for CP of a parent span the algorithm has to descend into children spans too. During that descent you can skip FOLLOWs links.

Reply:
> Yes, FOLLOWS span can have regular CHILD spans.
> 
> Shouldn't your algorithm be recursive? E.g. when we ask for CP of a parent span the algorithm has to descend into children spans too. During that descent you can skip FOLLOWs links.

I made the algorithm recursive
https://github.com/jaegertracing/jaeger-ui/blob/82a250a28b2dcd8d6ad625906cff4328542e3661/packages/jaeger-ui/src/components/TracePage/CriticalPath/utils/getChildOfSpans.tsx#L17-L22